shared/actions/chat2/index.tsx

```diff
@@ -2557,7 +2557,7 @@ const mobileNavigateOnSelect = (
     logger.info(
       `mobileNavigateOnSelect: navigating to selected: selected: ${state.chat2.selectedConversation} param: ${conversationIDKey}`
     )
-    return navigateToThreadRoute(state.chat2.selectedConversation, navKey)
+    return navigateToThreadRoute(conversationIDKey, navKey)
```

Comment:
seems dangerous to take the store value here. we don't do it in the other branches and i don't see why we'd ever want to

Reply:
Did this fix anything? Under what conditions could these values be different? I remember adding this log line because I thought it would be probably be a big deal if the parameter and the selected weren't the same.

Reply:
it shouldn't but its possible they can be divergent. the intention is to use the param so why ignore it

Reply:
I think the way forward might just be to get rid of the store's notion of the selected conversation. If they aren't the same I think that is guaranteed bad behavior, since some parts of the app are looking for the store version and some are looking for this other version flowing through these sagas. Assuming the navigation framework is internally consistent; that is, when we query for its current state it could always tell us with accuracy which conversation is selected, it seems like that is the right answer.